openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,51 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: block_structure.invalidate_cache_on_publish
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, the block structure cache is invalidated when changes to
+#   courses are published. Enabling this toggle is useful for non-expiring block structure cache.
+#   It is also useful to enable this when `block_structure.storage_backing_for_cache` is enabled
+#   because it guarantees the storage, containing the storage-backed structures, is updated with
+#   the latest changes.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
 INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
     "block_structure.invalidate_cache_on_publish", __name__
 )
+
+# .. toggle_name: block_structure.storage_backing_for_cache
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, block structures are stored in a more permanent storage,
+#   like a database, which provides an additional backup for cache misses, instead having them
+#   regenerated. The regenration of block structures is a time consuming processs. Therefore,
+#   enabling this switch is recommended for Production.
+#   For an example with additional context, see the following forum post:
+#   https://discuss.openedx.org/t/help-please-very-slow-load-time-10-seconds-for-courseware-on-sections-with-several-subsections-and-xblocks/2998/16
```

Comment:
Sorry.  The quality error, `36: C0301: (line-too-long), : Line too long (148/120)`, is my fault.  
This stinks. Not ideal, but maybe add a pylint disable pragma at the end of the line for now. 😞 

@regisb: Would it make sense to strip pylint disable pragmas, or maybe specific ones like this one, from the annotation text?

Reply:
Yeah, expected that error to be honest. I should have added the disable line too long lint.

True, it isn't an ideal solution, so hopefully @regisb will be able to help 😄 

For the time being, addressed 👍🏼

Reply:
I think we should not be adding links to discussion forums in the source code. Instead, these links should go to the git commit messages. Otherwise we would be adding tons of links (to Slack, Jira, forums) everywhere.

Reply:
@regisb I addressed your comment and moved the link from the annotation to the commit description 👍🏼 cc @robrap 

Please note that I squashed and force pushed so that it can be merged.